### DISCUSSION TITLE
Force stacks to restack after failed upload the first time

### DISCUSSION THREAD

**Discussion by dlang123** [simulot/immich-go#167] [Q&A]
So I tried to upload like 200k images using immich-go in one go. It hit a bunch of errors along the way and failed stacking and creating albums.  I ran it a second time and it uploaded 100k more images and created the albums but also failed stacking.  Ran it third time, it uploaded a few more images and updated some albums but didn't do any stacking.

Question:  Is there a way to force it to just stack images?

**Comment by simulot** [comment]
What kind of error have you got?

The is a command `stack` that will query the immich server and determine which images can be stacked.
https://github.com/simulot/immich-go#command-stack

**Reply by dlang123** [reply]
I get a lot of the following if I run the stack command and try to accept them.

Server status: OK
Get server's assets... 158668 received, 24432 stack(s) possible
Stack following images taken on 1998-01-04 03:12:00 -0600 CST
  006_6.JPG
  006_6.JPG
Can't stack images: GetAssetByID, GET, http://192.168.10.100:8080/api/asset/assetById/c82a8b1e-53a1-4833-847f-71439f6d0e58, 404 Not Found
Not Found

Stack following images taken on 1998-01-04 03:17:00 -0600 CST
  009_9.JPG
  009_9.JPG
Can't stack images: GetAssetByID, GET, http://192.168.10.100:8080/api/asset/assetById/bb5b6275-7d34-49d5-84ad-e9347fdc50a4, 404 Not Found
Not Found

I also verified that both files do exist:
/photos/library/name/1998/01/04/006_6.JPG
/photos/library/name/1998/01/04/006_6+1.JPG

now then, they are technically almost the same image and probably should be deleted, but this is just an example of all the 404 errors I saw

**Reply by simulot** [reply]
Can you test this for me:
1- open the immich page
2- duplicate the immich tab and change the url for one of the errors, example; http://192.168.10.100:8080/api/asset/assetById/c82a8b1e-53a1-4833-847f-71439f6d0e58

What the server replies?

**Reply by dlang123** [reply]
Get a 404 error, which says to me that its missing on the server somehow.  What confuses me is that I used Immich-go to upload all the pictures, although I had to run it 3 times in a row to get them all up there.

Error:
{
    "message": "Cannot GET /api/asset/assetById/c82a8b1e-53a1-4833-847f-71439f6d0e58",
    "error": "Not Found",
    "statusCode": 404
}

I also passed in Yes to the stack command.  

Get server's assets... 158667 received, 24434 stack(s) possible

24434 failed Stacks.

**Reply by simulot** [reply]
That's weird. The IDs come from the database, she are supposed to exist.

Could you describe your set up? Are using the an external library?

**Reply by simulot** [reply]
Wait, I have reproduced it!

**Reply by simulot** [reply]
Finally, that was another problem in my attempt to follow immich's API changes.

So back on your set up:
Are you using the external library feature? 
Do you have photos archived? trashed?

**Reply by dlang123** [reply]
None of the above. There might be one photo in trash. I just totally uploaded everything fresh install and I’m scared to put any time into it u til I know I’ve done every thing I can to maintain stacks and dates.

**Reply by peng1can** [reply]
Not sure if this helps, but I'm getting the same sort of errors:
```
Stacking PXL_20210408_221955753.MP.jpg, PXL_20210408_221955753.MP.jpg...
Can't stack images: GetAssetByID, GET, https://photos.xxxx.net/api/asset/assetById/d2e33cfe-6727-4905-82a9-4edceb8d065a, 404 Not Found
Not Found
```
This was on my first run importing an unzipped Google Takeout directory with about 40K assets.  This was a fresh immich install using no external libraries and the only configuration change I made at all was enabling oauth support.

**Reply by simulot** [reply]
I think I have found the problem.
The 0.10.0 version if immich-go uses an API entry GetAssetByID that will be soon deprecated. Apparently it don't work anymore and this breaks immich-go.

This will be fixed soon

**Reply by dlang123** [reply]
Thanks for the 0.11.0, it seems to have fixed many of the issues, but I still occassionally get a :
Stack following images taken on 2010-03-06 10:53:39.7 -0600 CST
  DSC_0499.JPG
  DSC_0499.JPG
Can't stack images: updateAssets, PUT, http://192.168.10.100:8080/api/asset, 400 Bad Request
Bad Request
each value in ids must be a UUID
ids should not be empty
ids must be an array

and these pop up occasionally.  I don't get as much info as we did in the past and I get one popping up probably every 10-20 stacks.

**Reply by simulot** [reply]
This smell as a bug. If you relaunch the task, does it fails at the same photo?
